runtime/src/accounts_db.rs

```diff
@@ -808,16 +810,32 @@ impl AccountsDB {
         Self::hash_account_data(
             slot,
             account.account_meta.lamports,
+            account.account_meta.executable,
+            account.account_meta.rent_epoch,
             account.data,
             &account.meta.pubkey,
         )
     }
 
     pub fn hash_account(slot: Slot, account: &Account, pubkey: &Pubkey) -> Hash {
-        Self::hash_account_data(slot, account.lamports, &account.data, pubkey)
+        Self::hash_account_data(
+            slot,
+            account.lamports,
```

Comment:
`lamports` and `rent_epoch` are both `u64`. So interchanging them caused no compile error. Spent some time to figure out why hashes are different. ;) So I added [another assertion](https://github.com/solana-labs/solana/pull/7415/files#diff-2099c5256db4eb5975c8834af38f6456R2182) covering that kind of bug.